echopop/graphics/plotting.py

```diff
@@ -0,0 +1,769 @@
+    )
+
+    # 'Hurdle' the data (separate presence/absence and magnitude processes)
+    data_hurdle = data_geo.loc[data_geo[variable] > 0.0].copy()
+    # ---- Sort by the variable magnitudes
+    data_hurdle = data_hurdle.sort_values(by=[variable]).reset_index(drop=True)
+    # ---- Normalize the point sizes
+    data_hurdle.loc[:, "normalized_var"] = scale_sizes(
+        data_hurdle.loc[:, variable], vmin, vmax, 1, 75
+    )
+
+    # Add the scatter plot
+    ax.scatter(
+        x=data_hurdle["longitude"],
+        y=data_hurdle["latitude"],
+        c=data_hurdle[variable],
+        norm=norm,
+        cmap=cmap,
+        s=data_hurdle["normalized_var"],
+        edgecolor="black",
+        linewidth=0.2,
+        zorder=plot_order,
+    )
+
+
+def plot_transect(
+    dataset: pd.DataFrame,
+    **kwargs,
+):
+
+    # Get the dataset variable name
+    variable = kwargs.get("variable", None)
+    # ---- Get the z-axis values
+    z = dataset[variable].values
+
+    # Get axis limits
+    if not kwargs.get("axis_limits", None):
+        # ---- Get survey bounds
+        survey_bounds = get_survey_bounds(dataset, kwargs.get("geo_config", None))
+        # ---- Additional buffering
+        axis_limits = dict(
+            x=dict(xmin=survey_bounds[0] * 1.005, xmax=survey_bounds[2] * 0.995),
+            y=dict(ymin=survey_bounds[1] * 0.995, ymax=survey_bounds[3] * 1.005),
+        )
+    else:
+        axis_limits = kwargs.get("axis_limits")
+
+    # Add prepend label string, if required
+    if "_male" in variable:
+        label_prepend = "Male "
+    elif "_female" in variable:
+        label_prepend = "Female "
+    else:
+        label_prepend = ""
+
+    # Prepare parameters for plotting
+    # ---- Abundance
+    if "abundance" in variable:
+        cmap = kwargs.get("cmap", "viridis")
+        colorbar_label = kwargs.get(
+            "colorbar_label", (label_prepend + "abundance\n$#~animals$").capitalize()
+        )
+    # ---- Biomass density
+    elif "biomass_density" in variable:
+        cmap = kwargs.get("cmap", "inferno")
+        colorbar_label = kwargs.get(
+            "colorbar_label",
+            (label_prepend + "biomass density\n$\\mathregular{kg~nmi^{-2}}$").capitalize(),
+        )
+    # ---- Biomass density
+    elif "biomass" in variable:
+        cmap = kwargs.get("cmap", "plasma")
+        colorbar_label = kwargs.get(
+            "colorbar_label", (label_prepend + "biomass\n$\\mathregular{kg}$").capitalize()
+        )
+    # ---- NASC
+    elif "nasc" in variable:
+        cmap = kwargs.get("cmap", "cividis")
+        colorbar_label = kwargs.get("colorbar_label", "NASC\n$\\mathregular{m^{2}~nmi^{-2}}$")
+    # ---- Number density
+    elif "number_density" in variable:
+        cmap = kwargs.get("cmap", "magma")
+        colorbar_label = kwargs.get(
+            "colorbar_label",
+            (label_prepend + "number density\n$\\mathregular{animals~nmi^{-2}}$").capitalize(),
+        )
```

Comment:
Since the variables are the same, seems like you can use a dict structure to store the default values for different variables and use the same `kwargs.get()` structure to substitute user-supplied values.

Same comments for the code under `plot_mesh`.